src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs

```diff
@@ -1,278 +0,0 @@
-        command.Visible = _suppressionStateService.CanSuppressSelectedEntries;
-        command.Enabled = command.Visible && !KnownUIContexts.SolutionBuildingContext.IsActive;
-    }
-
-    private void OnRemoveSuppressionsStatus(object sender, EventArgs e)
-    {
-        var command = (MenuCommand)sender;
-        command.Visible = _suppressionStateService.CanRemoveSuppressionsSelectedEntries;
-        command.Enabled = command.Visible && !KnownUIContexts.SolutionBuildingContext.IsActive;
-    }
-
-    private void OnAddSuppressionsInSourceStatus(object sender, EventArgs e)
-    {
-        var command = (MenuCommand)sender;
-        command.Visible = _suppressionStateService.CanSuppressSelectedEntriesInSource;
-        command.Enabled = command.Visible && !KnownUIContexts.SolutionBuildingContext.IsActive;
-    }
-
-    private void OnAddSuppressionsInSuppressionFileStatus(object sender, EventArgs e)
-    {
-        var command = (MenuCommand)sender;
-        command.Visible = _suppressionStateService.CanSuppressSelectedEntriesInSuppressionFiles;
-        command.Enabled = command.Visible && !KnownUIContexts.SolutionBuildingContext.IsActive;
-    }
-
-    private void OnAddSuppressionsInSource(object sender, EventArgs e)
-        => _suppressionFixService.AddSuppressions(selectedErrorListEntriesOnly: true, suppressInSource: true, projectHierarchy: null);
-
-    private void OnAddSuppressionsInSuppressionFile(object sender, EventArgs e)
-        => _suppressionFixService.AddSuppressions(selectedErrorListEntriesOnly: true, suppressInSource: false, projectHierarchy: null);
-
-    private void OnRemoveSuppressions(object sender, EventArgs e)
-        => _suppressionFixService.RemoveSuppressions(selectedErrorListEntriesOnly: true, projectHierarchy: null);
-
-    private void OnErrorListSetSeveritySubMenuStatus(object sender, EventArgs e)
-    {
-        // For now, we only enable the Set severity menu when a single configurable diagnostic is selected in the error list
-        // and we can update/create an editorconfig file for the configuration entry.
-        // In future, we can enable support for configuring in presence of multi-selection. 
-        var command = (MenuCommand)sender;
-        var selectedEntry = TryGetSingleSelectedEntry();
-        command.Visible = selectedEntry != null &&
-            !SuppressionHelpers.IsNotConfigurableDiagnostic(selectedEntry) &&
-            TryGetPathToAnalyzerConfigDoc(selectedEntry, out _, out _);
-        command.Enabled = command.Visible && !KnownUIContexts.SolutionBuildingContext.IsActive;
-    }
-
-    private void SetSeverityHandler(object sender, EventArgs args)
-    {
-        var selectedItem = (MenuCommand)sender;
-        var reportDiagnostic = TryMapSelectedItemToReportDiagnostic(selectedItem);
-        if (reportDiagnostic == null)
-        {
-            return;
-        }
-
-        var selectedDiagnostic = TryGetSingleSelectedEntry();
-        if (selectedDiagnostic == null)
```

Comment:
when TryGetSingleSelectedEntry returns null, this whole method is a no op.  So it doesn't call SetSeverityHandlerAsync, which unwinds everything.